### PR TITLE
Fix exceptions from pre-commit

### PR DESCRIPTION
**Pull request by hillaryj**
Fixes all the pre-commit error messages - not sure why it didn't run before pre-commit, but it's there now!

## 🗣 Description

Lots of small changes, mostly EOF and for shellcheck.

## 💭 Motivation and Context

Fixes the automation that we rely on.

## 🧪 Testing

Tested that these files load correctly on my local terminal.

## 🚥 Types of Changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: bash/.bashrc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+# shellcheck disable=1090
```
Comment (jsf9k) [inline bash/.bashrc]: Is it possible to disable just at the line where it is needed, instead of globally?

File: deploy.sh [hunk 2]
```diff
@@ -6,9 +6,11 @@ set -o pipefail
 
 echo "Finding dotfiles to deploy via stow..."
 
-for d in $(find . -maxdepth 1 -mindepth 1 -type d -not -name ".git" -exec basename {} \;)
```
Comment (jsf9k) [inline deploy.sh:9]: Why doesn't pre-commit like this `for` loop?  I think this is a clearer way of doing it than using a heredoc.

File: zsh/.zshrc.d/prompt [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #!/bin/zsh
 # shellcheck shell=bash
+# shellcheck disable=SC2034
```
Comment (jsf9k) [inline zsh/.zshrc.d/prompt]: Don't disable this globally.  Just disable it on the lines where it is needed.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I have a few issues/questions.  Please see my comments.

File: deploy.sh (hunk 2 above)
Comment (hillaryj) [reply deploy.sh:9]: Shellcheck recommends not using `for` loops due to fragility - this is one of their recommended POSIX-compliant patterns from https://github.com/koalaman/shellcheck/wiki/SC2044

I would also be happy to revert and just `disable=SC2044` if y'all prefer.

File: zsh/.zshrc.d/prompt (hunk 3 above)
Comment (hillaryj) [reply zsh/.zshrc.d/prompt]: I'll remove it entirely now that I've excluded the `zsh` files.

Comment (hillaryj) [reply zsh/.zshrc.d/prompt]: Fixed in https://github.com/cisagov/.dotfiles/pull/2/commits/e4d2a6d4a48c1b66d6ff25dfc8baf93d808fe876

File: bash/.bashrc (hunk 1 above)
Comment (hillaryj) [reply bash/.bashrc]: Refined in https://github.com/cisagov/.dotfiles/pull/2/commits/e4d2a6d4a48c1b66d6ff25dfc8baf93d808fe876

File: deploy.sh (hunk 2 above)
Comment (jsf9k) [reply deploy.sh:9]: I guess they have a point, even if the result is a little ugly.  In your case, though, I _think_ you can completely get rid of the `for` loop:
```
find . -maxdepth 1 -mindepth 1 -type d -not -name ".git" -exec stow -v -R -t ~ $(basename {}) \;
```

Does that work?

Comment (hillaryj) [reply deploy.sh:9]: I got a note that `This will expand once before find runs, not per file found`

Comment (jsf9k) [reply deploy.sh:9]: I think that message is a red herring.  Enclosing `$(basename {})` in quotes seems to do the trick.  See commit c2cf909, and let me know if you're OK with that.

**Review by jsf9k — APPROVED** [review]

File: bash/.bashrc.d/prompt [hunk 4]
```diff
@@ -12,23 +14,23 @@ RESET=$(tput sgr0)
 YELLOW=$(tput setaf 3)
 
 LINECHAR="─"
-PREFIX="❱" # U+2771
+PREFIX_CHAR="❱" # U+2771
```
Comment (dav3r) [inline bash/.bashrc.d/prompt:17]: Style thang here, but `LINECHAR` should probably now be renamed to `LINE_CHAR` to match `PREFIX_CHAR` and the rest of the variables in here.

**Review by dav3r — APPROVED** [review]
LGTM!  🏎️ 
Please consider my one minor comment.

File: deploy.sh (hunk 2 above)
Comment (hillaryj) [reply deploy.sh:9]: Looks great! I've tested it as well and it works well. Thanks!

File: bash/.bashrc.d/prompt (hunk 4 above)
Comment (hillaryj) [reply bash/.bashrc.d/prompt:17]: Resolved in https://github.com/cisagov/.dotfiles/pull/2/commits/069150a9663c4302f2804952169be9bc2e2c2fe6

**Pull request merged**